### PR TITLE
Create contact_rollups_pardot_memory table

### PR DESCRIPTION
**Pull request by bencodeorg**
Migration to create contact_rollups_pardot_memory table.

One immediate question -- I have an index constraining `pardot_id` (and email, for that matter) to be unique. In our design doc, we have planned on adding a `deleted_at` column to allow soft deletes -- I think that could cause problems if we soft delete a record with email benj@minbrooks.com and try and create a record with the same email later. Should I just remove the uniqueness constraint?

## Testing story

Executed migration locally.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      t.datetime :pardot_id_updated_at
+      t.json :data_synced
+      t.datetime :data_synced_at
+      t.datetime :data_rejected_at
+      t.string :data_rejected_reason
+      t.timestamps
+    end
+
+    add_index :contact_rollups_pardot_memory, :email, unique: true
+    add_index :contact_rollups_pardot_memory, :pardot_id, unique: true
```
Comment (hacodeorg) [inline dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb:15]: How does this work for null-able field?

**Review by hacodeorg — COMMENTED** [review]
We don't have to worry about soft-deleting yet. It's just a non-critical improvement for a later time. The unique index on email is okay.

File: dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb (hunk 1 above)
Comment (bencodeorg) [reply dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb:15]: [MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/create-index.html) say it allows multiple null values:

> A UNIQUE index creates a constraint such that all values in the index must be distinct. An error occurs if you try to add a new row with a key value that matches an existing row. If you specify a prefix value for a column in a UNIQUE index, the column values must be unique within the prefix length. A UNIQUE index permits multiple NULL values for columns that can contain NULL.

Comment (hacodeorg) [reply dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb:15]: I see. This index could be helpful to quickly validate new data and find the highest Pardot ID.

Comment (hacodeorg) [reply dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb:15]: FYI, the mysql link above is version 8.0 while the version we have in Aurora is 5.7.12. The part we care about is mostly the same between 2 versions.

**Review by hacodeorg — APPROVED** [review]

File: dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb (hunk 1 above)
Comment (bencodeorg) [reply dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb:15]: Ah, good catch thanks!

**Pull request merged**